include/EASTL/bonus/ring_buffer.h

```diff
@@ -1516,7 +1521,7 @@ namespace eastl
 		const typename ring_buffer<T, Container, Allocator>::size_type sizeB = b.size();
 
 		if(sizeA == sizeB)
-			return (a.c < b.c);
+			return eastl::lexicographical_compare(a.begin(), a.end(), b.begin(), b.end());
 		return sizeA < sizeB;
```

Comment:
`lexicographical_compare` should handle different size ranges, I think for consistency with other containers (like vector), this should just return the same as the lexicographical comparison. So a ring buffer with `{ 1, 2 }` in it compares smaller to one with `{ 3 }` in it (as it would for vectors, for example)

Reply:
Done.